paig-securechat/web-server/src/paig_securechat/services/langchain_service_intf.py

```
@@ -63,15 +64,14 @@ def _get_conversational_retrieval_chain(self, conversation_messages=None, temper
 
         try:
             qa = ConversationalRetrievalChain.from_llm(self.langchain_llm, memory=memory, verbose=True,
-                                                           retriever=self.vectordb.as_retriever(
-                                                               search_kwargs={"k": self.vector_store_retriever_k}),
-                                                           return_source_documents=self.return_source_documents,
-                                                           response_if_no_docs_found=self.response_if_no_docs_found)
-
+                                                       retriever=self.vectordb.as_retriever(
+                                                           search_kwargs={"k": self.vector_store_retriever_k}),
+                                                       return_source_documents=self.return_source_documents,
+                                                       response_if_no_docs_found=self.response_if_no_docs_found)
             return qa
-        except Exception as e:
-            logger.error(f"Exception occurred while getting qa chain with error :: {e}")
-            raise e
+        except Exception:
+            logger.error("Failed to create QA chain due to an internal error.")
+            return None
```

why are we returning none instead of raising an exception ?

---

i did that because
If an exception is raised, it must be handled elsewhere in the code.
Returning None allows the caller function to check the return value and handle it as needed instead of dealing with exceptions.

---

Should We Raise an Exception Instead? As you suggest

---

stack trace revealing to user ? In logs or UI ?

---

In logs , there was some confusion regarding the task which was fixed in the Wednesday meeting with Prvaeen Sir and Siddheshwar sir